src/Persistence/PersistentObjectFactory.php

```diff
@@ -277,20 +296,21 @@ protected function normalizeParameter(string $field, mixed $value): mixed
             if ($inversedRelationshipMetadata && !$inversedRelationshipMetadata->isCollection) {
                 $inverseField = $inversedRelationshipMetadata->inverseField;
 
-                // we create now the object to prevent "non-nullable" property errors,
-                // but we'll need to remove it once the current object is created
-                $inversedObject = unproxy($value->create());
-                $this->tempAfterPersist[] = static function(object $object) use ($value, $inverseField, $pm, $inversedObject) {
-                    // we cannot use the already created $inversedObject:
-                    // because we must also remove its potential newly created owner (here: "$oldObj")
-                    // but a cascade:["persist"] would remove too many things
-                    $value->create([$inverseField => $object]);
-                    $pm->refresh($object);
-                    $oldObj = get($inversedObject, $inverseField);
-                    delete($inversedObject);
-                    if ($oldObj) {
-                        delete($oldObj); // @phpstan-ignore argument.templateType
-                    }
+                // we need to handle the circular dependency involved by inversed one-to-one relationship:
+                // a placeholder object is used, which will be replaced by the real object, after its instantiation
+                $inversedObject = $value->withoutPersistingButScheduleForInsert()
```

Comment:
we're creating the "child" object here, but it needs the object created by the current factory, and we're passing it an empty "placeholder". So we should not persist now, or it would break. 
Otherwise, we should "schedule for insert" the "child" object, as well as all other objects that would be created by the factory `$value`. 

Hence this "persist" mode stuff